packages/a2/bgl/src/a2/utils.ts

```diff
@@ -29,9 +29,7 @@ export {
   decodeBase64,
 };
 
-export type NonPromise<T> = T extends Promise<unknown> ? never : T;
-
-function ok<T>(o: Outcome<NonPromise<T>>): o is NonPromise<T> {
+function ok<T>(o: Outcome<Awaited<T>>): o is Awaited<T> {
```

Comment:
Is this right? I just wanted to make sure that you couldn't put promises as argument (a common problem I ran into). Will this do the same?

Reply:
Also, this change is a no-op in A2. These files are write-only: they are written out for readability of the code that is stored in BGL.

Reply:
Yes, Awaited does that and it is better, because it unroll type type recursively. e.g. Awaited<Promise<Promise<T>> = T

Oh I forgot about A2, I'm reverting this change, it's not necessary, the one I need is in packages/breadboard/src/data/file-system/utils.ts